proguard/proguard-project.pro

```diff
@@ -0,0 +1,34 @@
+-verbose
+
+# We should at least keep the Exceptions, InnerClasses,
+# and Signature attributes when processing a library.
+-keepattributes Exceptions,InnerClasses,Signature,RuntimeVisibleParameterAnnotations,RuntimeVisibleAnnotations
+
+# Preserve all native method names and the names of their classes.
+-keepclasseswithmembernames,includedescriptorclasses class * {
+    native <methods>;
+}
+
+# Preserve the special static methods that are required in all enumeration classes.
+-keepclassmembers enum * {
+    public static **[] values();
+    public static ** valueOf(java.lang.String);
+}
+
+-keepclassmembernames class * {
+    java.lang.Class class$(java.lang.String);
+    java.lang.Class class$(java.lang.String, boolean);
+}
+
+-keepclassmembers class * implements java.io.Serializable {
+    static final long serialVersionUID;
+    static final java.io.ObjectStreamField[] serialPersistentFields;
+    private void writeObject(java.io.ObjectOutputStream);
+    private void readObject(java.io.ObjectInputStream);
+    java.lang.Object writeReplace();
+    java.lang.Object readResolve();
+}
+
+-keepclassmembers class * implements android.os.Parcelable {
+    android.os.Parcelable$Creator CREATOR;
+}
```

Comment:
Curious 🤔 Why do we need all these rules? Is this because of Kotlin?

Reply:
actually I've added a few more than here should be. Remove unnecessary 👍